lib/clone_kit.rb

```diff
@@ -24,13 +26,18 @@ def self.spec
     @spec ||= {}
   end
 
+  def self.reset_graph!
+    @graph = nil
+  end
+
   def self.add_specification(specification)
     spec[specification.model.name] = specification
     refresh_specification(specification)
   end
 
   def self.refresh_specification(specification)
     graph.add_vertex(specification.model.name, *specification.dependencies)
+    specification.cloner.register_id_generator_with_rules
```

Comment:
Why do we need to call this here?  It seems like we could just call `rule.id_generator = id_generator` from within the loop of `apply_rules_and_save`.